tests/unit/test_req_install.py

```diff
@@ -42,3 +45,50 @@ def test_forward_slash_results_in_a_link(self, tmpdir):
         )
 
         assert requirement.link is not None
+
+
+class TestInstallRequirementFrom(object):
+
+    def test_install_req_from_string_without_comes_from(self):
+        """
+        Test to make sure that install_req_from_string succeeds
+        when called with URL (PEP 508) but without comes_from.
+        """
+        # Test with a PEP 508 url install string:
+        wheel_url = ("https://download.pytorch.org/whl/cu90/"
+                     "torch-1.0.0-cp36-cp36m-win_amd64.whl")
+        install_str = "torch@ " + wheel_url
+        install_req = install_req_from_req_string(install_str)
+
+        assert isinstance(install_req, InstallRequirement)
+        assert install_req.link.url == wheel_url
+        assert install_req.req.url == wheel_url
+        assert install_req.comes_from is None
+        assert install_req.is_wheel
+
+    def test_install_req_from_string_with_comes_from_without_link(self):
+        """
+        Test to make sure that install_req_from_string succeeds
+        when called with URL (PEP 508) and comes_from
+        does not have a link.
+        """
+        # Test with a PEP 508 url install string:
+        wheel_url = ("https://download.pytorch.org/whl/cu90/sds"
```

Comment:
I think you did not mean to add the extra `sds` ?

Reply:
@xavfernandez, removed.  Thanks for the catch that got introduced during styling.